lib/galaxy/jobs/actions/post.py

```diff
@@ -427,7 +427,7 @@ def execute(cls, app, sa_session, action, job, replacement_dict):
                 else:
                     log.debug("No job found yet for wfi_step %s, (step %s)" % (wfi_step, wfi_step.workflow_step))
             for j2c in jobs_to_check:
-                creating_jobs = [(x, x.dataset.creating_job) for x in j2c.input_datasets if x.dataset.creating_job]
+                creating_jobs = [(x, x.dataset.creating_job) for x in j2c.input_datasets if x.dataset and x.dataset.creating_job]
```

Comment:
@nsoranzo Sure, I can do that.  Someone will need to actually follow up on the debug output though ;)